src/Hl7.Fhir.Base/Serialization/FhirXmlException.cs

```diff
@@ -29,58 +34,79 @@ public class FhirXmlException : CodedException
-        internal static readonly FhirXmlException ELEMENT_OUT_OF_ORDER = new(ELEMENT_OUT_OF_ORDER_CODE, "Element '{0}' is not in the correct order ");
-        internal static readonly FhirXmlException ELEMENT_NOT_IN_SEQUENCE = new(ELEMENT_NOT_IN_SEQUENCE_CODE, "Element with name '{0}' was found multiple times, but not in sequence.");
-        internal static readonly FhirXmlException UNALLOWED_ELEMENT_IN_RESOURCE_CONTAINER = new(UNALLOWED_ELEMENT_IN_RESOURCE_CONTAINER_CODE, "Encountered unallowed content '{0}' in the resource container. Only a single resource is allowed.");
-        internal static readonly FhirXmlException NO_ATTRIBUTES_ALLOWED_ON_RESOURCE_CONTAINER = new(NO_ATTRIBUTES_ALLOWED_ON_RESOURCE_CONTAINER_CODE, "The element '{0}' has a contained resource and therefore should not have attributes.");
-        internal static readonly FhirXmlException UNALLOWED_NODE_TYPE = new(UNALLOWED_NODE_TYPE_CODE, "Xml node of type '{0}' is unexpected at this point");
-        internal static readonly FhirXmlException INCORRECT_ATTRIBUTE_NAMESPACE = new(INCORRECT_ATTRIBUTE_NAMESPACE_CODE, "The attribute '{0}' in element '{1}' uses the namespace '{2}', which is not allowed.");
-        internal static readonly FhirXmlException ATTRIBUTE_HAS_EMPTY_VALUE = new(ATTRIBUTE_HAS_EMPTY_VALUE_CODE, "Attributes cannot be empty. Either they are absent, or they are present with at least one character of non - whitespace content");
-        internal static readonly FhirXmlException SCHEMALOCATION_DISALLOWED = new(SCHEMALOCATION_DISALLOWED_CODE, "The 'schemaLocation' attribute is disallowed.");
-        internal static readonly FhirXmlException EXPECTED_OPENING_ELEMENT = new(EXPECTED_OPENING_ELEMENT_CODE, "Expected opening element, but found {0}.");
-        internal static readonly FhirXmlException ENCOUNTERED_DTD_REFERENCES = new(ENCOUNTERED_DTP_REFERENCES_CODE, "There SHALL be no DTD references in FHIR resources (because of the XXE security exploit)");
-        internal static readonly FhirXmlException ELEMENT_HAS_NO_VALUE_OR_CHILDREN = new(ELEMENT_HAS_NO_VALUE_OR_CHILDREN_CODE, "Element '{0}' must have child elements and / or a value attribute");
+        // ==========================================
+        // Unrecoverable Errors (or data loss)
+        // ==========================================
+        internal static FhirXmlException UNKNOWN_RESOURCE_TYPE(XmlReader reader, string instancePath, string typeName) => Initialize(reader, instancePath, UNKNOWN_RESOURCE_TYPE_CODE, $"Unknown type '{typeName}' found in root property.", OO_Sev.Fatal, OO_Typ.Structure);
+        internal static FhirXmlException RESOURCE_TYPE_NOT_A_RESOURCE(XmlReader reader, string instancePath, string resourceType) => Initialize(reader, instancePath, RESOURCE_TYPE_NOT_A_RESOURCE_CODE, $"Data type '{resourceType}' in property 'resourceType' is not a type of resource.", OO_Sev.Fatal, OO_Typ.Structure);
```

Comment:
This list is longer, see https://github.com/FirelyTeam/firely-net-sdk/blob/ae47a158854646bb210b0f135ddbce5d639d5e2d/src/Hl7.Fhir.Base/Serialization/FhirXmlException.cs